public/app/api/clients/folder/v1beta1/hooks.ts

```diff
@@ -202,27 +206,85 @@ export function useDeleteMultipleFoldersMutationFacade() {
     return deleteFolders;
   }
 
-  return async function deleteFolders({ folderUIDs }: { folderUIDs: string[] }) {
+  return async function deleteFolders({ folderUIDs }: DeleteFoldersArgs) {
+    const successMessage = t('folders.api.folder-deleted-success', 'Folder deleted');
+
     // Delete all the folders sequentially
     // TODO error handling here
     for (const folderUID of folderUIDs) {
       // This also shows warning alert
-      if (await isProvisionedFolderCheck(dispatch, folderUID)) {
-        continue;
+      const isProvisioned = await isProvisionedFolderCheck(dispatch, folderUID);
+
+      if (!isProvisioned) {
+        const result = await deleteFolder({ name: folderUID });
+        if (!result.error) {
+          // Before this was done in backend srv automatically because the old API sent a message wiht 200 request. see
+          // public/app/core/services/backend_srv.ts#L341-L361. New API does not do that so we do it here.
+          getAppEvents().publish({
+            type: AppEvents.alertSuccess.name,
+            payload: [successMessage],
+          });
+        }
       }
-      const result = await deleteFolder({ name: folderUID });
-      if (!result.error) {
-        // Before this was done in backend srv automatically because the old API sent a message wiht 200 request. see
-        // public/app/core/services/backend_srv.ts#L341-L361. New API does not do that so we do it here.
-        getAppEvents().publish({
-          type: AppEvents.alertSuccess.name,
-          payload: [t('folders.api.folder-deleted-success', 'Folder deleted')],
+    }
+
+    dispatch(refreshParents(folderUIDs));
+    return { data: undefined };
+  };
+}
+
+export function useMoveMultipleFoldersMutationFacade(): readonly [
+  (args: MoveFoldersArgs) => Promise<{ data?: unknown }>,
+  ReturnType<typeof useMoveFoldersMutationLegacy>[1] | ReturnType<typeof useUpdateFolderMutation>[1],
+] {
```

Comment:
You can remove the types here and instead return the app platform hook as this - it then should get rid of the type problems you were encountering:

```tsx
return [moveFolders, updateFolderData] as const;
```